### ISSUE TITLE
[java] SingularField doesn't consider anonymous classes defined in non-private fields

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#3620]
**Affects PMD Version:** 6.40.0

**Rule:** [SingularField](https://pmd.github.io/latest/pmd_rules_java_design.html#singularfield)

**Description:** The rule skips all non-private fields, but these fields could define a nested anonymous class which itself has singular fields.

**Code Sample demonstrating the issue:**

```java
public class Foo {
    Object field = new Object() {
        private int x; // singular field

        int bar(int y) {
            x = y + 5;
            return x;
        }
    };
}
```

**Expected outcome:**

PMD should report a violation at line 3, but doesn't. This is a false-negative.
